rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,763 @@
+  # convenience.
+  /v1/catalogs/{catalog}:
+    parameters:
+      - name: catalog
+        in: path
+        required: true
+        description: Name of the catalog being configured
+        schema:
+          type: string
+          minLength: 1
+    post:
+      tags:
+         - Configuration API
+      summary: Persist catalog specific configuration, which can be retrieved
+        for later use.
+      operationId: postConfig
+      description:
+        Persist some catalog specific configurations, which will be returned by \
+        calls to /v1/config in the future. This is basically all of the data \
+        that would go into the Catalog's `initialize` call.
+      # TODO - Make this into a CatalogConfiguration
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Catalog'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized / Invalid Credentials Provided
+
+  # Could also consider /v1/tables/{identifier}
+  /v1/namespaces/{namespace}/tables/{table}:
+    parameters:
+      # Consider moving this to query parameters,so it can be more easily URL encoded
+      # in case of dots or special characters
+      - name: namespace
+        in: path
+        description: Namespace the table is in
+        required: true
+        schema:
+          type: string
+        examples:
+          singlepart_namespace:
+            value: "prod"
+          multipart_namespace:
+            value: "prod.accounting"
+      - name: table
+        in: path
+        description: Name of the table to load
+        required: true
+        schema:
+          type: string
+        example: "sales"
+    get:
+      tags:
+        - Catalog API
+      summary: Load a given table from a given namespace
+      operationId: loadTable
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetTableResponse'
+        '401':
+          description: Unauthorized
+        # Using 412, `Precondition Failed`, instead of 404, as 404 makes monitoring response codes from ELBs
+        #  very difficult - Hard to tell if clients or servers are misconfigured and calling non-existent routes
+        #  or missing routes versus expected error cases such as NoSuchTableException (expected meaning that
+        #  a person who is on call shouldn't be paged for this but 404 they might need to be).
+        '412':
+          description: NoSuchTableException
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchTableError'
+    put:
+      tags:
+        - Catalog API
+      summary: Commit an in progress create (or replace) table transaction
+      operationId: commitTable
```

Comment:
Can you remove this operation? This is part of `TableOperations`, not `Catalog` and I don't think we want to dive into table updates in this PR.

Reply:
So what I was trying to cover here was the `createTableTransaction` APIs, which are part of `Catalog`.

However, since the actual commit isn't part of `Catalog`, it's probably best to not include it yet since we're just covering `Catalog`. Will remove.

Reply:
Removed!